glmmbernoulli.qmd

```diff
@@ -0,0 +1,523 @@
+$0\le\mu\le 1$.
+
+In general we don't want to have restrictions on the values of the linear predictor so we equate the linear predictor to a function of $\mu$ that has an unrestricted range.
+In the case of the Bernoulli distribution with the canonical link function we equate the linear predictor to the *log odds* or *logit* of the positive response.
+That is
+$$
+\eta = \log\left(\frac{\mu}{1-\mu}\right) .
+$$ {#eq-logodds}
+
+To understand why this is called the "log odds" recall that $\mu$ corresponds to a probability in $[0,1]$.
+The corresponding odds ratio, $\frac{\mu}{1-\mu}$, is in $[0,\infty)$ and the logarithm of the odds ratio, $\mathrm{logit}(\mu)$, is in $(-\infty, \infty)$.
+
+The inverse of the logit link function,
+$$
+\mu = \frac{1}{1+\exp(-\eta)} ,
+$$ {#eq-logitinv}
+is called the *logistic* function and is shown in @fig-logitinv.
+
+```{julia}
+#| code-fold: true
+#| fig-cap: 'The logistic function, which is the inverse to the logit link function.'
+#| label: fig-logitinv
+let
+  fig = Figure(; resolution=(800, 400))
+  ax = Axis(fig[1, 1]; xlabel="η", ylabel="μ")
+  lines!(ax, -5.5 .. 5.5, η -> inv(1 + exp(-η)))
+  fig
+end
+```
+
+The inverse link takes a value on the unrestricted range, $(-\infty,\infty)$, and maps it to the probability range, $[0,1]$.
+It happens this function is also the cumulative distribution function for the standard logistic distribution, available in [Distributions.jl](https://github.com/JuliaStats/Distributions.jl) as `cdf(Logistic(), η)`.
+In some presentations the relationship between the logit link and the logistic distribution is emphasized but that often leads to questions of why we should focus on the logistic distribution.
+Also, it is not clear how this approach would generalize to other distributions such as the Poisson or the Gamma distributions.
+
+### Canonical link functions {#sec-canonicallink}
+
+A way of deriving the logit link that does generalize to a class of common distributions, in what is called the *exponential family*, is to consider the logarithm of the probability function (for discrete distributions) or the probability density function (for continuous distributions).
+The probability function for the Bernoulli distribution is $\mu$ for $y=1$ and $1-\mu$ for $y=0$.
+If we write this in a somewhat contrived way as $\mu^y+(1-\mu)^{1-y}$ for $y\in\{0,1\}$ then the logarithm of the probability function becomes
+$$
+\log\left(\mu^y+(1-\mu)^{1-y}\right) = \log(1-\mu) +
+y\,\log\left(\frac{\mu}{1-\mu}\right) .
+$$ {#eq-BernoulliProb}
+Notice that the logit link function is the multiple of $y$ in the last term.
+
+The characteristic of distributions in the exponential family is that the logarithm of the probability mass function or probability density function, whichever is appropriate, can be expressed as a sum of up to three terms: one that involves $y$ only, one that involves the parameters only and the product of $y$ and a function of the parameters.
```

Comment:
```suggestion
The characteristic of distributions in the exponential family is that the logarithm of the probability mass function or probability density function, whichever is appropriate, can be expressed as a sum of up to three terms: one that involves $y$ only, one that involves the parameters only, and the product of $y$ and a function of the parameters.
```

I'm not normally a fan of the Oxford comma, but it does help a little when you complex entities in your list.

By the wy, OUP official style guide holds the same opinion of the Oxford comma as I do, which makes me really curious why it's called the Oxford comma...